src/components/Editor/AddFileModal.tsx

```diff
@@ -0,0 +1,71 @@
+import { Dialog } from '@headlessui/react';
+import { useSetAtom } from 'jotai';
+import React, { useContext, useRef, useState } from 'react';
+import { createNewInternalSolutionFileAtom } from '../../atoms/editor';
+import { DarkModeContext } from '../../context/DarkModeContext';
+import { AlgoliaEditorSolutionFile } from '../../models/algoliaEditorFile';
+import Modal from '../Modal';
+import Select from '../Select';
+import parse from './parsers/parse';
+export default function AddFileModal(props) {
+  const darkMode = useContext(DarkModeContext);
+  const [division, setDivision] = useState('');
+  const [fileStatus, setFileStatus] = useState('Create File');
+  const fileURLRef = useRef<HTMLInputElement>(null);
+  const createSol = useSetAtom(createNewInternalSolutionFileAtom);
+  console.log(division);
+  return (
+    <Modal {...props}>
+      <Dialog.Panel className="bg-white dark:bg-black w-full max-w-xl dark:text-white p-5 rounded-lg shadow-lg flex flex-col items-start">
+        <h3 className="text-lg font-bold">Enter Problem URL</h3>
+        <input
+          type="url"
+          className="shadow-sm focus:ring-blue-500 focus:border-blue-500 block w-full sm:text-sm border-gray-300 rounded-md dark:bg-gray-900 dark:border-gray-700"
+          placeholder="e.g. https://codeforces.com/contest/1920/problem/C"
+          ref={fileURLRef}
+        />
```

Comment:
why ref and not controlled input (ie. `const [fileURL, setFileURL] = useState('')`?)

Reply:
Not sure, is there a significant difference? I always thought useRef on inputs would be faster, since typing wouldn't trigger a re-render.

Reply:
I don't think the performance overhead will be that high... I think React generally prefers `useState`: https://stackoverflow.com/questions/76086163/what-is-better-usestate-or-useref-when-handling-multiple-form-inputs-in-react

Reply:
got it 👍🏻